go/vt/vttablet/tabletmanager/vreplication/table_plan_builder.go

```diff
@@ -402,53 +409,59 @@ func (tpb *tablePlanBuilder) findCol(name sqlparser.ColIdent) *colExpr {
 func (tpb *tablePlanBuilder) generateInsertStatement() *sqlparser.ParsedQuery {
 	bvf := &bindvarFormatter{}
 	buf := sqlparser.NewTrackedBuffer(bvf.formatter)
-	if tpb.onInsert == insertIgnore {
-		buf.Myprintf("insert ignore into %v set ", tpb.name)
+
+	tpb.generateInsertPart(buf)
+	if tpb.lastpk == nil {
+		buf.Myprintf(" values ", tpb.name)
+		tpb.generateValuesPart(buf, bvf)
 	} else {
-		buf.Myprintf("insert into %v set ", tpb.name)
-	}
-	tpb.generateInsertValues(buf, bvf)
-	if tpb.onInsert == insertOndup {
-		buf.Myprintf(" on duplicate key update ")
-		tpb.generateUpdate(buf, bvf, false /* before */, true /* after */)
+		tpb.generateSelectPart(buf, bvf)
 	}
+	tpb.generateOnDupPart(buf)
+
 	return buf.ParsedQuery()
 }
 
-func (tpb *tablePlanBuilder) generateUpdateStatement() *sqlparser.ParsedQuery {
+func (tpb *tablePlanBuilder) generateInsertPart(buf *sqlparser.TrackedBuffer) *sqlparser.ParsedQuery {
 	if tpb.onInsert == insertIgnore {
-		return tpb.generateInsertStatement()
+		buf.Myprintf("insert ignore into %v(", tpb.name)
+	} else {
+		buf.Myprintf("insert into %v(", tpb.name)
 	}
-	bvf := &bindvarFormatter{}
-	buf := sqlparser.NewTrackedBuffer(bvf.formatter)
-	buf.Myprintf("update %v set ", tpb.name)
-	tpb.generateUpdate(buf, bvf, true /* before */, true /* after */)
-	tpb.generateWhere(buf, bvf)
+	separator := ""
+	for _, cexpr := range tpb.colExprs {
+		buf.Myprintf("%s%s", separator, cexpr.colName.String())
+		separator = ","
+	}
+	buf.Myprintf(")", tpb.name)
 	return buf.ParsedQuery()
 }
 
-func (tpb *tablePlanBuilder) generateDeleteStatement() *sqlparser.ParsedQuery {
-	bvf := &bindvarFormatter{}
-	buf := sqlparser.NewTrackedBuffer(bvf.formatter)
-	switch tpb.onInsert {
-	case insertNormal:
-		buf.Myprintf("delete from %v", tpb.name)
-		tpb.generateWhere(buf, bvf)
-	case insertOndup:
-		buf.Myprintf("update %v set ", tpb.name)
-		tpb.generateUpdate(buf, bvf, true /* before */, false /* after */)
-		tpb.generateWhere(buf, bvf)
-	case insertIgnore:
-		return nil
+func (tpb *tablePlanBuilder) generateValuesPart(buf *sqlparser.TrackedBuffer, bvf *bindvarFormatter) *sqlparser.ParsedQuery {
+	bvf.mode = bvAfter
```

Comment:
A comment here will be helpful, something to the effect that setting the mode will control how Myprintf format's the bind variable.

Reply:
Since these are set in multiple places, I've added a detailed comment on the `bindvarFormatter` type.